pipeline/src/convert_csv.ts

```diff
@@ -0,0 +1,77 @@
+import { program } from "commander";
+import util from "util";
+
+import FS from 'node:fs';
+import assert from 'node:assert';
+import {parse} from 'csv-parse/sync';
+
+import fs = require("node:fs/promises");
+import path = require("node:path");
+
+import { Table, filterTables, mergeTablesByModelNumber } from "./table_merger";
+import { APPLIANCE_TYPES } from "../../backend/schema/metadata";
+import { glob } from "glob";
+import { retrieveMetadata } from "./metadata";
+
+const SPECS_FILE_BASE = "../data/";
+const INPUT_SUBDIR = "reformatted/";
+const OUTPUT_SUBDIR = "";
+const RUNS = "runs/";
+
+program.parse();
+
+async function main() {
+// loop through the sub folders in the run folder
+const rfolder = path.join(SPECS_FILE_BASE, RUNS).replace(/\\/g, '/');
+FS.readdir(rfolder, async (err, topFolders) => {
+    if (err || !topFolders.length){
+        console.log(`No runs detected`);
+    }
+    else{
+        for (const file of topFolders){
+            // Try to open the "all_records.csv" file
+            const runFile = path.join(rfolder, file, "all_records.csv");
+            console.log(`Trying ${runFile}`);
+            FS.readFile(runFile, function (err, content) {
+                // Skip the folder if there is no file
+                if(err){
+                    console.log(`Skipping ${file} because it is empty`);
+                }
+                else{
+                    //Parse content and convert to JSON
+                    console.log(`Parsing file found in ${runFile}`);
+                    const records = parse(content, {
+                        skip_empty_lines: true,
+                        columns: true
+                    });
+                    // Create raw data file
+                    const rawFileName = path.join(rfolder, file, "all_records.JSON").replace(/\\/g, '/');
+                    const recordVals = Object.values(records);
+                    const jsonFormatRecords = JSON.stringify(recordVals);
+
+                    fs.writeFile(rawFileName, jsonFormatRecords, "utf-8");
+
+                    // Create file for each type of appliance JSON key: "applianceType"
+                    // "applianceType"s key values pulled from the metadata file
+                    for (const appliance of Object.values(APPLIANCE_TYPES)){
```

Comment:
There's a nice library called lodash that's already installed in the `pipeline` that has a groupBy method that would make this easier.

https://www.geeksforgeeks.org/lodash-_-groupby-method/

You'd basically call `_.groupBy(records, (appliances) => appliance["applianceType"]`, which would give you a grouped object with the appliance type as key and the records as the values. You could then loop through the APPLIANCE_TYPES and if they exist in the grouped object, you'd write them to a file.

It's not strictly necessary but I think would be cleaner.

Reply:
Used filter instead of group so that only the desired entries were returned